src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -183,6 +183,9 @@ function ($m) {
 
         restore_error_handler();
 
+        // Remove NULL characters to prevent DOMDocument::loadHTML() from stopping the document parsing
+        $content = str_replace("\x00", '', $content);
```

Comment:
I would instead throw an exception. Modifying the input without warning the user does not seem like a good idea to me.